pts.go

```diff
@@ -89,27 +89,32 @@ func (p PTS) RolledOver(other PTS) bool {
 	return false
 }
 
-// DurationFrom returns the difference between the two pts times. This number is always positive.
-func (p PTS) DurationFrom(from PTS) uint64 {
-	switch {
-	case p.RolledOver(from):
-		return uint64((MaxPtsTicks - from) + p)
-	case from.RolledOver(p):
-		return uint64((MaxPtsTicks - p) + from)
-	case p < from:
-		return uint64(from - p)
-	default:
-		return uint64(p - from)
+// DurationFrom returns the difference between the two pts times.
+// One possible distance has rollover and wraps around the number line,
+// the other one does not wrap around the number line and doesn't have rollover.
+// DurationFrom will assume the smallest difference is the correct one.
+// This number is always positive.
+// This function will only be accurate if the difference between the two
+// PTS values is less than MaxPtsTicks/2 (2^32 = 4294967296 = 0x100000000)
+func (p PTS) DurationFrom(from PTS) PTS {
+	var difference PTS
+	if p == PtsPositiveInfinity || p == PtsNegativeInfinity || from == PtsPositiveInfinity || from == PtsNegativeInfinity {
+		return PtsPositiveInfinity
+	}
+	if p > from {
+		difference = p - from
+	} else {
+		difference = from - p
 	}
+	if difference >= MaxPtsTicks/2 {
+		difference = MaxPtsTicks - difference
+	}
+	return difference
 }
 
 // Add adds the two PTS times together and returns a new PTS
 func (p PTS) Add(x PTS) PTS {
-	result := p + x
-	if result > MaxPtsValue {
-		result = result - MaxPtsTicks
-	}
-	return PTS(result)
+	return (p + x) & MaxPtsValue
```

Comment:
This is a good change, it actually fixes an issue if the caller ever sent in a `value > MaxPtsValue` which is possible since there the type checking is against uint64, not something like uint33.  It may be wise to use an arbitrary precision library to define our own 33 bit unsigned integer so we get some nice static type checking.

In the mean time it may be wise to add a unit test showing that it still works if the caller sends in large values.